### ISSUE TITLE
Aria activedescendant on SVG not allowed

### ISSUE THREAD

**Issue by udacityanna** [dequelabs/axe-core#2170]
Hi! I check my company's browser app for 508 compliance. I submitted a bug from Axe to our devs, but the lead dev said it's invalid. We probably have an uncommon use, so I thought it _could_ be a bug in the rule set. Otherwise, I could use some help figuring out what is going on. It's an error that is repeated many times and I'm not ready to just dismiss it.

But checking https://www.w3.org/TR/html-aria/#index-aria-application, https://www.w3.org/TR/wai-aria-1.1/#aria-activedescendant, and https://html.spec.whatwg.org/multipage/dom.html#flow-content, it seems like the Axe error _is_ invalid. I used the browser version of Axe (Axe core 3.5) in preparation for incorporating more automated scans. I was evaluating libraries.

Our web app is not open, so I can't post a link. There are many svg graphs used on data dashboards. Keyboard users can navigate to each one and additionally tab through each one as a table. This has passed testing by users of AT. Is it that it is considered not accessibility-supported?

Here are screenshots:
<img width="1201" alt="Screen Shot 2020-04-03 at 6 02 13 PM copy" src="https://user-images.githubusercontent.com/12027531/79793371-72699100-831e-11ea-835d-95b5ab21f940.png">
<img width="623" alt="Screen Shot 2020-04-20 at 3 39 56 PM" src="https://user-images.githubusercontent.com/12027531/79793385-77c6db80-831e-11ea-9f01-12f2a40cb84d.png">

Expectation: << Describe what you expect the product to do >>

Actual: << Describe what the product actually does >>

Motivation: << Describe why you want the behavior to be changed >>

<pre><code>
axe-core version: X.Y.Z
axe-webdriver, extension or other integration version: X.Y.Z

Browser and Assistive Technology versions

For Tooling issues:
- Node version: XX  
- Platform:  
</code></pre>

**Comment by straker** [comment]
So SVG does not support `aria-activedescendant` as a valid attribute. The [attribute is only allowed on](https://www.w3.org/TR/wai-aria-1.1/#aria-activedescendant) `application`, `composite`, `group`, and `textbox` roles (and subroles) and an [SVG is none of those roles](https://www.w3.org/TR/html-aria/). 

Looking at the code and image I'm not sure what the intended purpose of the attribute is in your use case.

**Comment by udacityanna** [comment]
Wow! Thank you for the reply! I would imagine the intent was to use the SVG as a role=application and that use the activedescendant. I will ask the lead dev for comment.

**Comment by jozefbriss** [comment]
It seems SVG can be role=application according to https://www.w3.org/TR/html-aria/#el-svg 

![image](https://user-images.githubusercontent.com/35486165/79830466-b9e73000-83a5-11ea-8253-32c412efeae5.png)

**Comment by straker** [comment]
Shoot, I missed the `role=application` in the code screenshot somehow. It seems our internal lookup table does not list `aria-activedescendant` as a valid option for `role=application`. Let me find out if this was intentional.

**Comment by straker** [comment]
Alright so `application` not having `aria-activedescendant` as a valid attribute seems to be incorrect on our end. We're also going to do some testing in various browsers and screen readers just to see how they treat it (as we are curious how it works).

**Comment by udacityanna** [comment]
I'm really grateful to be able to get to the bottom of this!

**Comment by WilcoFiers** [comment]
@udacityanna I just wanted to comment how much I love the different patterns in that coloured pie chart! That's very cool.

**Comment by udacityanna** [comment]
@WilcoFiers That compliment really means a lot! That particular part of our charts and graphs was difficult to design. Our game is programmed to need 12 unique colors and we were trying to make it meet contrast (there is a hover event), but also be good for color blind users, and also not be too busy.

**Comment by straker** [comment]
Did some research with this structure:

```html
<div role="application" tabindex="0" aria-activedescendant="button1">
  <button id="button1">Cut</button>
  <button id="button2">Copy</button>
  <button id="button3">Paste</button>
</div>
```

Firefox/NVDA and IE11/JAWS read the button name when focusing the outer div. However, Safari/VoiceOver only read that a button is active and not its name. Testing shows that `aria-activedescendant` is supported in those browsers those.

**Comment by padmavemulapati** [comment]
Verified with the `<div role="application" tabindex="0" aria-activedescendant="button1">
  <button id="button1">Cut</button>
  <button id="button2">Copy</button>
  <button id="button3">Paste</button>
</div>`
and with axe-core3.5.5 its passing the aria-allowed-attribute for 'role="application" tabindex="0" aria-activedescendant="button1"`

![image](https://user-images.githubusercontent.com/47409579/85304820-d26ce700-b4c9-11ea-9c2c-07eac103f61f.png)

**Comment by jeankaplansky** [comment]
no user-facing docs required.
